### ISSUE TITLE
[Bug] KeyError: 'CLASSES'

### ISSUE THREAD

**Issue by vansin** [open-mmlab/mmrotate#680]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmrotate/issues) and [Discussions](https://github.com/open-mmlab/mmrotate/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmrotate.readthedocs.io/en/1.x/notes/4_faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmrotate) or [latest version (1.x)](https://github.com/open-mmlab/mmrotate/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

1.x branch https://github.com/open-mmlab/mmrotate/tree/1.x

### Environment

```shell
(mmroate) ➜  mmrotate git:(qd) ✗ python mmrotate/utils/collect_env.py 
sys.platform: linux
Python: 3.8.15 (default, Nov 24 2022, 15:19:38) [GCC 11.2.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: Tesla V100-SXM2-32GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.4, V11.4.152
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.10.0+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.0+cu111
OpenCV: 4.6.0
MMEngine: 0.4.0
MMRotate: 1.0.0rc0+3db9cc0
```

### Reproduces the problem - code sample

```shell
1.5225  loss_cls: 0.8153  loss_bbox: 0.7072
01/02 17:39:40 - mmengine - INFO - Saving checkpoint at 1 epochs
Traceback (most recent call last):
  File "tools/train.py", line 122, in <module>
    main()
  File "tools/train.py", line 118, in main
    runner.train()
  File "/home/ubuntu/mmroate-1.x/mmengine/mmengine/runner/runner.py", line 1678, in train
    model = self.train_loop.run()  # type: ignore
  File "/home/ubuntu/mmroate-1.x/mmengine/mmengine/runner/loops.py", line 96, in run
    self.runner.val_loop.run()
  File "/home/ubuntu/mmroate-1.x/mmengine/mmengine/runner/loops.py", line 347, in run
    self.run_iter(idx, data_batch)
  File "/home/ubuntu/miniconda3/envs/mmroate/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 28, in decorate_context
    return func(*args, **kwargs)
  File "/home/ubuntu/mmroate-1.x/mmengine/mmengine/runner/loops.py", line 368, in run_iter
    self.evaluator.process(data_samples=outputs, data_batch=data_batch)
  File "/home/ubuntu/mmroate-1.x/mmengine/mmengine/evaluator/evaluator.py", line 60, in process
    metric.process(data_batch, _data_samples)
  File "/home/ubuntu/mmroate-1.x/mmrotate/mmrotate/evaluation/metrics/dota_metric.py", line 287, in process
    for label in range(len(self.dataset_meta['CLASSES'])):
KeyError: 'CLASSES'
```

### Reproduces the problem - command or script

```shell
python tools/train.py configs/rotated_retinanet/rotated-retinanet-rbox-le90_r50_fpn_1x_dota.py
```

### Reproduces the problem - error message

```shell
python tools/train.py configs/rotated_retinanet/rotated-retinanet-rbox-le90_r50_fpn_1x_dota.py
```

### Additional information

need I propose a Pull Request to replace CLASSES to classes in https://github.com/open-mmlab/mmrotate/blob/b88863dd258a8bae63560e9f97b1b2e088cba36e/mmrotate/evaluation/metrics/dota_metric.py#L287

**Comment by zytx121** [comment]
Please check mmdet >= `3.0.0rc5`.

**Comment by vansin** [comment]
yes, use the latest mmdet dev-3.x

**Comment by zytx121** [comment]
Fix it in 1.0.0rc1
